tests/js/lintAll-tests.js

```diff
@@ -0,0 +1,90 @@
+"use strict";
+var fluid = require("infusion");
+var jqUnit = require("node-jqunit");
+
+require("../../src/js/lint-all");
+
+fluid.registerNamespace("fluid.test.lintAll");
+
+// TODO: There is never output from one of the tests, and there's no summary.  Discuss with Antranig.
+
+fluid.test.lintAll.checkSingleTally = function (tally) {
+    fluid.each(["checked", "invalid", "valid"], function (key) {
+        var singleTallyEntry = fluid.get(tally, key);
+        jqUnit.assertTrue("Each tally entry should be a number.", typeof singleTallyEntry === "number");
+        jqUnit.assertTrue("Each tally entry should be positive.", singleTallyEntry > 0);
+    });
+
+    jqUnit.assertTrue("The tally entries should match.", tally.checked === (tally.valid + tally.invalid));
+};
+
+fluid.test.lintAll.checkSingleResult = function (testDef) {
+    fluid.log(fluid.logLevel.FAIL, testDef.message);
+    jqUnit.asyncTest(testDef.message, function () {
+        jqUnit.expect(testDef.shouldFail ? 1 : 70); // If the run succeeds, there are 10 checks * 7 tests
+
+        var allChecksPromise = fluid.lintAll.runAllChecks(testDef.argsOptions);
```

Comment:
This is why there's no output, the tests are not actually run because the function doesn't exist.  First I'll ensure that it fails properly (because of the expect levels set above) and then actually write the danged function.

Reply:
See above, the tests actually run now and I fixed the issues revealed.